servers/rendering/renderer_rd/storage_rd/texture_storage.h

```diff
@@ -290,13 +289,45 @@ class TextureStorage : public RendererTextureStorage {
 		RS::ViewportVRSMode vrs_mode = RS::VIEWPORT_VRS_DISABLED;
 		RID vrs_texture;
 
+		// overridden textures
+		struct RTOverridden {
+			RID color;
+			RID depth;
+			RID velocity;
+
+			struct SliceKey {
+				RID rid;
+				uint32_t layer = 0;
+
+				bool operator==(const SliceKey &p_val) const {
+					return (rid == p_val.rid) && (layer == p_val.layer);
+				}
+
+				static uint32_t hash(const SliceKey &p_val) {
+					uint32_t h = hash_one_uint64(p_val.rid.get_id());
+					h = hash_murmur3_one_32(p_val.layer, h);
+					return hash_fmix32(h);
+				}
+
+				SliceKey() {}
+				SliceKey(RID p_rid, uint32_t p_layer) {
+					rid = p_rid;
+					layer = p_layer;
+				}
+			};
+
+			mutable HashMap<SliceKey, RID, SliceKey> cached_slices;
```

Comment:
Is this just to make it easier to support and arbitrary number of views? It kind of looks like overkill for most use cases

Reply:
Keep in mind that we're cycling through multiple texture chains that are not controlled by our rendering device but created in this case by OpenXR, so we don't know how many entries we need to keep track off. It's not just the number of layers which we could limit to 2 for now, it's the number of entries in each texture chain. So I had to keep it as generic as possible.

Reply:
Ah, that is a huge pain. Could you put a comment to that effect so future contributors understand the complexity is there for a reason?

Reply:
Added some info above the struct :)